command/agent/http.go

```diff
@@ -531,13 +531,7 @@ func (s *HTTPServer) parseToken(req *http.Request, token *string) {
 // DC in the request, if given, or else the agent's DC.
 func (s *HTTPServer) parseSource(req *http.Request, source *structs.QuerySource) {
 	s.parseDC(req, &source.Datacenter)
-	if node := req.URL.Query().Get("near"); node != "" {
```

Comment:
Does it break? This just pushes the check out to the `consul` side instead of the `agent`. I was trying to avoid checking the magic flag in the agent and on the consul side (since the `Near` parameter can live in either place now).